community-containers/lldap/lldap.json

```diff
@@ -0,0 +1,81 @@
+      ],
+      "nextcloud_exec_commands": [
+        "php /var/www/html/occ app:install user_ldap",
+        "php /var/www/html/occ app:enable user_ldap",
+        "php /var/www/html/occ ldap:create-empty-config",
+        "php /var/www/html/occ ldap:set-config s01 ldapAgentName 'uid=ro_admin,ou=people,%NC_BASE_DN%'",
+        "php /var/www/html/occ ldap:set-config s01 ldapAgentPassword '%OPENLDAP_LDAP_USER_PASS%'",
+        "php /var/www/html/occ ldap:set-config s01 ldapBase '%NC_BASE_DN%'",
+        "php /var/www/html/occ ldap:set-config s01 ldapBaseGroups '%NC_BASE_DN%'",
+        "php /var/www/html/occ ldap:set-config s01 ldapBaseUsers '%NC_BASE_DN%'",
+        "php /var/www/html/occ ldap:set-config s01 ldapCacheTTL 600",
+        "php /var/www/html/occ ldap:set-config s01 ldapConfigurationActive 1",
+        "php /var/www/html/occ ldap:set-config s01 ldapEmailAttribute 'mail'",
+        "php /var/www/html/occ ldap:set-config s01 ldapExperiencedAdmin 0",
+        "php /var/www/html/occ ldap:set-config s01 ldapGidNumber 'gidNumber'",
+        "php /var/www/html/occ ldap:set-config s01 ldapGroupDisplayName 'cn'",
+        "php /var/www/html/occ ldap:set-config s01 ldapGroupFilter '(&(objectclass=groupOfUniqueNames))'",
+        "php /var/www/html/occ ldap:set-config s01 ldapGroupFilterGroups ''",
+        "php /var/www/html/occ ldap:set-config s01 ldapGroupFilterMode 0",
+        "php /var/www/html/occ ldap:set-config s01 ldapGroupFilterObjectclass 'groupOfUniqueNames'",
+        "php /var/www/html/occ ldap:set-config s01 ldapGroupMemberAssocAttr 'uniqueMember'",
+        "php /var/www/html/occ ldap:set-config s01 ldapHost 'nextcloud-aio-openldap'",
+        "php /var/www/html/occ ldap:set-config s01 ldapLoginFilterAttributes 'uid'",
+        "php /var/www/html/occ ldap:set-config s01 ldapLoginFilterEmail 0",
+        "php /var/www/html/occ ldap:set-config s01 ldapLoginFilterUsername 1",
+        "php /var/www/html/occ ldap:set-config s01 ldapMatchingRuleInChainState 'unknown'",
+        "php /var/www/html/occ ldap:set-config s01 ldapNestedGroups 0",
+        "php /var/www/html/occ ldap:set-config s01 ldapPagingSize 500",
+        "php /var/www/html/occ ldap:set-config s01 ldapPort 3890",
+        "php /var/www/html/occ ldap:set-config s01 ldapTLS 0",
+        "php /var/www/html/occ ldap:set-config s01 ldapUserAvatarRule 'default'",
+        "php /var/www/html/occ ldap:set-config s01 ldapUserDisplayName 'displayname'",
+        "php /var/www/html/occ ldap:set-config s01 ldapUserFilter'(&(objectClass=person)(uid=%uid))'",
+        "php /var/www/html/occ ldap:set-config s01 ldapUserFilterMode 1",
+        "php /var/www/html/occ ldap:set-config s01 ldapUserFilterObjectclass 'person'",
+        "php /var/www/html/occ ldap:set-config s01 ldapUuidGroupAttribute 'auto'",
+        "php /var/www/html/occ ldap:set-config s01 ldapUuidUserAttribute 'auto'",
+        "php /var/www/html/occ ldap:set-config s01 turnOnPasswordChange 0"
+      ]
```

Comment:
I fear this autoconfig is not going to work yet since some preparations (allow to define nextcloud_variables,
 allow to define nextcloud_secrets) are missing. This is tracked in https://github.com/nextcloud/all-in-one/discussions/3889#discussion-5931541. We could however document these commands in the readme for which it is there :)

Reply:
I understand. I would like if we add ldap to be as simple as possible.

So, I can try to implement the missing functionalities.

Reply:
> I understand. I would like if we add ldap to be as simple as possible.
> 
> So, I can try to implement the missing functionalities.

That is a great offer, thanks a lot! However after thinking about it, the nextcloud_exec_commands are usually executed every restart of the nextcloud container which is probably not what we want here. So I guess there would additionally need to be an option to allow to execute these commands in this case only once. Not sure yet how this could be implemented.

Lets for the first iteration add the commands above to the readme like suggested and do additionall stuff in a follow-up PR. Does this sound good to you?